claims.go

```diff
@@ -140,174 +122,63 @@ func securityLifeCycleToString(v uint16) string {
 	return "invalid"
 }
 
-// FromCOSE unwraps a Claims from the supplied CWT (For now only
-// COSE-Sign1 wrapping is supported.)
-func (p *Evidence) FromCOSE(cwt []byte) error {
-	if !cose.IsSign1Message(cwt) {
-		return errors.New("not a COSE-Sign1 message")
-	}
-
-	p.message = cose.NewSign1Message()
-
-	err := p.message.UnmarshalCBOR(cwt)
-	if err != nil {
-		return err
-	}
+func isPSAHashType(b []byte) error {
+	l := len(b)
 
-	// To verify the CWT we need to locate the verification key
-	// which is identified by the InstanceID claim inside the PSA
-	// token.
-	err = p.FromCBOR(p.message.Payload)
-	if err != nil {
-		return err
+	if l != 32 && l != 48 && l != 64 {
```

Comment:
A cost names is better then such a complex digit based evaluation!

Reply:
mmm is that really complex? seems quite simple to me :-) I prefer transparency rather than hiding the values behind a necessarily more opaque const.